common/components/Header/components/NetworkSelector/NetworkSelector.tsx

```diff
@@ -0,0 +1,227 @@
+      suffix = node.name;
+    } else if (node.isAuto) {
+      // Auto nodes should show the count of all nodes it uses. If only one,
+      // show the service name of the node.
+      const networkNodes = Object.values(allNodes).filter(
+        n => !isAutoNodeConfig(n) && n.network === node.network
+      );
+
+      if (networkNodes.length > 1) {
+        suffix = `${networkNodes.length} Nodes`;
+      } else {
+        suffix = networkNodes[0].service;
+      }
+    } else {
+      suffix = node.service;
+    }
+
+    return (
+      <span>
+        {network.name} <small>({suffix})</small>
+      </span>
+    );
+  };
+
+  private renderOptions = () => {
+    const { allNodes, allNetworks, nodeSelection } = this.props;
+    const { expandedNetwork, isShowingAltNetworks } = this.state;
+    const selectedNode = allNodes[nodeSelection];
+
+    const nodesByNetwork = {} as {
+      [network: string]: NodeConfig[];
+    };
+    Object.values(allNodes).forEach((node: NodeConfig) => {
+      if (!nodesByNetwork[node.network]) {
+        nodesByNetwork[node.network] = [];
+      }
+      nodesByNetwork[node.network].push(node);
+    }, {});
+
+    const options = {
+      core: [] as React.ReactElement<any>[],
+      alt: [] as React.ReactElement<any>[]
+    };
+    Object.keys(nodesByNetwork)
+      .sort((a, b) => {
+        // Sort by CORE_NETWORKS first, custom networks last
+        const idxA = CORE_NETWORKS.includes(a) ? CORE_NETWORKS.indexOf(a) : 999;
+        const idxB = CORE_NETWORKS.includes(b) ? CORE_NETWORKS.indexOf(b) : 999;
+        return idxA - idxB;
+      })
+      .forEach(netKey => {
+        const network = allNetworks[netKey];
+        const nodeType = CORE_NETWORKS.includes(netKey) || network.isCustom ? 'core' : 'alt';
+        options[nodeType].push(
+          <NetworkOption
+            key={netKey}
+            network={allNetworks[netKey]}
+            nodes={nodesByNetwork[netKey]}
+            nodeSelection={nodeSelection}
+            isNetworkSelected={selectedNode.network === netKey}
+            isExpanded={expandedNetwork === allNetworks[netKey]}
+            selectNetwork={this.selectNetwork}
+            selectNode={this.selectNode}
+            toggleExpand={this.toggleNetworkExpand}
+          />
+        );
+      });
+
+    return (
+      <div className="NetworkSelector">
+        {options.core}
+        <button className="NetworkSelector-alts" onClick={this.toggleShowAltNetworks}>
+          <i className="fa fa-flask" />
+          {translate(isShowingAltNetworks ? 'HIDE_THING' : 'SHOW_THING', {
+            $thing: translateRaw('NETWORKS_ALTERNATIVE')
+          })}
+        </button>
+        {isShowingAltNetworks && options.alt}
+        <button className="NetworkSelector-add" onClick={this.openCustomNodeModal}>
+          <i className="fa fa-plus" />
+          {translate('NODE_ADD')}
+        </button>
+      </div>
+    );
+  };
+
+  private selectNetwork = (net: NetworkConfig) => {
```

Comment:
e.g. the function signature would be `(net: NetworkConfig) => () = > void)`. And then in the rendering function call `selectNetwork(allNetworks[netKey])`. That way you can just call it in `NetworkOption` without creating a wrapper function inside it